src/ShallowWaterRosenbrock.jl

```diff
@@ -0,0 +1,252 @@
+function shallow_water_rosenbrock_time_step!(
+     y₂, ϕ, F, q₁, q₂, duh₁, duh₂, H1h, H1hchol, y_wrk,  # in/out args
+     model, dΩ, dω, Y, V, Q, R, S, f, g, y₁, yₘ,         # in args
+     RTMMchol, L2MMchol, Amat, Bchol, Blfchol,           # more in args
+     dt, τ, leap_frog)                                   # ...yet more in args
+  # energetically balanced second order rosenbrock shallow water solver
+  # reference: eqns (24) and (39) of
+  # https://github.com/BOM-Monash-Collaborations/articles/blob/main/energetically_balanced_time_integration/EnergeticallyBalancedTimeIntegration_SW.tex
+  #
+  # f          : coriolis force (field)
+  # g          : gravity (constant)
+  # h₁         : fluid depth at current time level
+  # u₁         : fluid velocity at current time level
+  # RTMM       : H(div) mass matrix, ∫β⋅βdΩ, ∀β∈ H(div,Ω)
+  # L2MM       : L² mass matrix, ∫γγdΩ, ∀γ∈ L²(Ω)
+  # dt         : time step
+  # τ          : potential vorticity upwinding parameter
+  # dΩ         : measure of the elements
+
+  n = get_normal_vector(model)
+  dt₁ = dt
+  if leap_frog
+    dt₁ = 2.0*dt
+  end
+
+  # multifield terms
+  u₁, h₁ = y₁
+  u₂, h₂ = y₂
```

Comment:
Move this line (L28) right before it is needed.

Reply:
This has been moved to the second step